src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -18,22 +18,26 @@ namespace Internal.Runtime.CompilerHelpers
     /// </summary>
     internal static class InteropHelpers
     {
+        private static unsafe byte* CharArrayToAnsi(char* pStr, int stringLength)
+        {
+            // CORERT-TODO: Use same encoding as the rest of the interop
+            var encoding = Encoding.UTF8;
+            int bufferLength = encoding.GetByteCount(pStr, stringLength);
+            byte* buffer = (byte*)PInvokeMarshal.CoTaskMemAlloc((UIntPtr)(void*)(bufferLength + 1)).ToPointer();
```

Comment:
Also, note that you need to have separate the marshaling and the allocation. There are cases you only marshal, but without allocation. Example: Native -> Managed with [out]. Also, in that case, you might also need to pass the native array length to avoid overwriting the buffer. Finally, the allocation needs to be of the right size, which isn't necessarily the ansi char array size.